src/mrb_capability.c

```diff
@@ -266,6 +294,87 @@ mrb_value mrb_cap_getgid(mrb_state *mrb, mrb_value self)
     return mrb_fixnum_value((mrb_int)getgid());
 }
 
+mrb_value mrb_file_cap_init(mrb_state *mrb, mrb_value self)
+{
+    char *path;
+    mrb_file_cap_context *cap_ctx = (mrb_file_cap_context *)mrb_malloc(mrb, sizeof(mrb_file_cap_context));
+
+    mrb_get_args(mrb, "z", &path);
+    cap_ctx->path = path;
+    cap_ctx->cap = cap_get_file(path);
+
+    mrb_iv_set(mrb
+        , self
+        , mrb_intern_cstr(mrb, "mrb_file_cap_context")
+        , mrb_obj_value(Data_Wrap_Struct(mrb
+            , mrb->object_class
+            , &mrb_file_cap_context_type
+            , (void *)cap_ctx)
+        )
+    );
```

Comment:
In current mruby, you should use DATA_PTR for wrapping struct data instread of `mrb_iv_set`. (We also should fix other function using `mrb_iv_set` like `mrb_cap_init`.